mac/Keyman4MacIM/Keyman4MacIM/KMDataRepository.m

```diff
@@ -0,0 +1,185 @@
+/**
+ * Keyman is copyright (C) SIL International. MIT License.
+ *
+ * KMResourcesRepository.m
+ * Keyman
+ *
+ * Created by Shawn Schantz on 2024-07-30.
+ *
+ * Singleton object which serves as an abstraction for the reading and writing of Keyman data.
+ * The 'data' currently consists of keyman keyboards installed by the user. All data is saved locally on disk using NSFileManager.
+ * This is in contrast with the lighter weight Settings which is stored using UserDefaults and handled by KMSettingsRepository.
+ */
+
+#import "KMDataRepository.h"
+#import "KMLogs.h"
+
+@interface KMDataRepository ()
+@property (readonly) NSURL *applicationSupportSubDirectory;   // '~/Library/Application Support'
+@property (readonly) NSURL *documentsSubDirectory;    // '~/Documents'
+@property (readonly) NSURL *obsoleteKeymanKeyboardsDirectory; // '~/Library/Documents/Keyman-Keyboards'
```

Comment:
The documentsSubDirectory is the user's Documents directory, identified using an API. The obsolete directory is the full path to the place we formerly stored data under the Documents directory.

Looking at the code, I can eliminate the property documentsSubDirectory as it is only used in finding the full obsolete data path. Would that help simplify this? It is a special location, so I thought it would be useful to save the reference, but we really never want to access it again after the migration.

I'll change the comments to further clarify.

Reply:
After working on the comments a little. I think I need a better comment that explains the hierarchical relationship between these. We have three directories we should know of going forward and two obsolete ones.

Reply:
Yes, sounds good, thanks